tools/rewriter/GenCallAsm.cpp

```diff
@@ -1026,6 +1039,19 @@ std::string emit_asm_wrapper(AbiSignature &sig,
 
   emit_set_return_pkru(aw, caller_pkey, arch);
 
+  // Call the post-condition function, if one was specified.
+  // The call happens in the caller's compartment.
+  // For now, we hardcode the existence and name of the post-condition function.
+  // The name is `${target_name}_post_condition`,
+  // and we only do this for `dav1d_get_picture`.
+  std::optional<std::string> target_post_condition_name = std::nullopt;
+  if (target_name && *target_name == "dav1d_get_picture") {
```

Comment:
Hardcoding this is fine, but I think this won't let us run the rewriter on the dav1d demo without a post-condition function. Since we're considering using it for the ARM demo and there will be more rewriter codegen changes like in #471 maybe we should gate the post-condition validation with a feature flag.

It can be either a build-time option (e.g. preprocessor #define) or a runtime option (e.g. extra rewriter flag), whichever is easiest. I'm also good with having it on by default to make integrating your changes into the test infra easier.

Reply:
I made it a runtime flag in eb8f17b; runtime seemed easier.  On by default, which seems simpler for now.  It'll hopefully soon get un-hardcoded, so I simplest seemed best here.

Reply:
thanks! that approach is good with me